dev_mode/package.json

```diff
@@ -211,7 +211,7 @@
     "yarn-deduplicate": "^5.0.0"
   },
   "engines": {
-    "node": ">=12.0.0"
```

Comment:
Maybe we could indeed update the Node version used on CI, but leave this one to `>=14.0.0`?

Reply:
or maybe we require Node 16 now.

Node 14 is already in maintenance according to https://nodejs.dev/en/about/releases/

![image](https://user-images.githubusercontent.com/591645/212091118-33e2eca2-9ab2-4810-befd-04814a3c46dd.png)

Reply:
Normally, long-term support (LTS) releases end support on April 30, so one would expect Node 16 to be in LTS until 2024-04-30. However, due to a dependency on a soon-to-be-EOL version of OpenSSL, Node 16 is ending LTS on 2023-09-11, just under 8 months from now. That's why I was recommending that we adopt 18 instead of 16.

Reply:
Right, it would be great to encourage folks to use Node 18 if they want to build JupyterLab.

The previous change happened in https://github.com/jupyterlab/jupyterlab/pull/9282, with a more conservative bump from 10 to 12 instead of going to 14 directly.

Potential concerns could be with repo2docker / Binder which usually have Node in the default environment. Linking to https://github.com/jupyterhub/repo2docker/pull/983 for reference.

Reply:
Although now end users are less likely to need Node on their machine. Before it was used more often because source extensions had to be installed with the `jupyter labextension install` command.

Reply:
Thinking about it more, updating to Node 18 directly sounds good, since this will be part of JupyterLab 4. Thanks @JasonWeill :+1: